# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: aef69344a652
# Full Hash: aef69344a6529cbb635d63ca29036f9565bf895f
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2022-01-29 09:17:08
# Description:
#   Bug 1751919 - Crash in [@ mozilla::dom::SessionHistoryEntry::ReplaceWith]. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137305
# ==============================================================================

diff -r ce0468b4f920 -r aef69344a652 docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Fri Jan 28 18:40:13 2022 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Fri Jan 28 18:52:21 2022 +0000
@@ -817,7 +817,7 @@
           mActiveEntry = newActiveEntry;
         } else if (LOAD_TYPE_HAS_FLAGS(
                        aLoadType, nsIWebNavigation::LOAD_FLAGS_IS_REFRESH) &&
-                   !ShouldAddEntryForRefresh(newActiveEntry)) {
+                   !ShouldAddEntryForRefresh(newActiveEntry) && mActiveEntry) {
           addEntry = false;
           mActiveEntry->ReplaceWith(*newActiveEntry);
         } else {
